ktor-server/ktor-server-core/src/io/ktor/features/ContentNegotiation.kt

```diff
@@ -73,32 +75,24 @@ class ContentNegotiation(val converters: List<ContentConverterRegistration>) {
             }
             return feature
         }
-
-        private fun PipelineContext<Any, ApplicationCall>.acceptedTypes(): List<Pair<ContentType, Double>> {
-            val acceptHeader = call.request.header(HttpHeaders.Accept)
-            val acceptHeaderValues = parseHeaderValue(acceptHeader)
-            return acceptHeaderValues
-                    .map { ContentType.parse(it.value) to it.quality }
-                    .sortedWith(contentTypeComparator)
-        }
     }
 }
 
 class UnsupportedMediaTypeException(contentType: ContentType) : Exception("Content type $contentType is not supported")
 
 interface ContentConverter {
-    suspend fun convertForSend(context: PipelineContext<Any, ApplicationCall>, value: Any): Any?
+    suspend fun convertForSend(context: PipelineContext<Any, ApplicationCall>, contentType: ContentType, value: Any): Any?
     suspend fun convertForReceive(context: PipelineContext<ApplicationReceiveRequest, ApplicationCall>): Any?
 }
 
-class ConvertedContent(text: String, val contentType: ContentType) : OutgoingContent.ByteArrayContent() {
-    private val bytes = text.toByteArray(Charsets.UTF_8)
+class ConvertedContent(val text: String, val contentType: ContentType, override val status: HttpStatusCode? = null) : OutgoingContent.ByteArrayContent() {
```

Comment:
It is not good at all: we should keep only one of them at a time: only text or only bytes: we shouldn't keep both as it wastes memory